services/cd-service/pkg/service/batch.go

```diff
@@ -0,0 +1,139 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
+
+Copyright 2021 freiheit.com*/
+package service
+
+import (
+	"context"
+	"fmt"
+	"github.com/freiheit-com/kuberpult/pkg/api"
+	"github.com/freiheit-com/kuberpult/services/cd-service/pkg/repository"
+	"github.com/freiheit-com/kuberpult/services/cd-service/pkg/valid"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+type BatchServer struct {
+	Repository repository.Repository
+}
+
+func (d *BatchServer) processAction(
+	batchAction *api.BatchAction,
+) (repository.Transformer , error) {
+	switch action := batchAction.Action.(type) {
+	case *api.BatchAction_CreateEnvironmentLock:
+		if !valid.EnvironmentName(action.CreateEnvironmentLock.Environment) {
+			return nil, status.Error(codes.InvalidArgument, "invalid environment")
```

Comment:
please improve these error messages.
e.g.
`return nil, status.Error(codes.InvalidArgument, fmt.sprintf("cannot create environment lock: invalid environment: '%s'", action.CreateEnvironmentLock.Environment))`
for this and all the other errors below.
This will help users of the api to identify problems much more easily.